custom_components/bosch_alarm/config_flow.py

```diff
@@ -50,13 +52,24 @@ async def try_connect(hass: HomeAssistant, data: dict[str, Any]):
     Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
     """
     panel = Panel(host=data[CONF_HOST], port=data[CONF_PORT],
-                  passcode=data[CONF_PASSWORD])
+                  automation_code=data[CONF_PASSWORD], installer_code=data.get(CONF_INSTALLER_CODE, None)) 
+    errors = {}
+
     try:
         await panel.connect(Panel.LOAD_BASIC_INFO)
+    except (PermissionError, ValueError):
+        errors["base"] = "invalid_auth"
```

Comment:
It's a bit less redundant to just `raise RuntimeError("invalid_auth")` here and construct `errors` at the caller